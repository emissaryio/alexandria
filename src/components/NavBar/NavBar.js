import React, { Component, Children } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import IconButton from '@material-ui/core/IconButton';
import HomeIcon from '@material-ui/icons/Home';
import Toolbar from '@material-ui/core/Toolbar';


class NavBar extends Component {
  state = {
    value: 0
  };

  handleChange = (event, value) => {
    this.setState({ value });
  };

  render() {
    const {
      children,
      className,
      color
    } = this.props;

    const {
      value
    } = this.state;


    return (
      <div
        className={classnames('', className)}
      >
        <AppBar position="static" color={color}>
          <Toolbar>
            <IconButton href="/">
              <HomeIcon />
            </IconButton>
            <Tabs
              value={value}
              onChange={this.handleChange}
              indicatorColor={(color === 'primary') ? 'secondary' : 'primary'}
              variant="fullWidth"
            >
              {Children.map(children, child => (
                <Tab label={child.props.label}>
                  {child}
                </Tab>
              ))}
            </Tabs>
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

NavBar.propTypes = {
  className: PropTypes.string,
  children: PropTypes.arrayOf(PropTypes.element).isRequired
};

export default NavBar;
