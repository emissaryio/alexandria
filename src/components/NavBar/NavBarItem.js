import React from 'react';
import Tab from '@material-ui/core/Tab';
import Link from '../Link';


const NavBarItem = ({ label, to }) => (
  <Tab label={label}>
    <Link to={to}>{label}</Link>
  </Tab>
);

export default NavBarItem;
