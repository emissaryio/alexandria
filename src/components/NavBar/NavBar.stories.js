import React from 'react';
import { storiesOf } from '@storybook/react';

import { withTests } from '@storybook/addon-jest';
import { MuiThemeProvider } from '@material-ui/core/styles';
import Theme from '../../hocs/withTheme';
import NavBar from './NavBar';
import NavBarItem from './NavBarItem';


storiesOf('NavBar', module)
  .addDecorator(withTests('NavBar'))
  .addWithJSX('default', () => (
    <MuiThemeProvider theme={Theme}>
      <NavBar color="primary">
        <NavBarItem label="Item 1" to="#Item1" />
        <NavBarItem label="Item 2" to="#Item2" />
        <NavBarItem label="Item 3" to="#Item3" />
      </NavBar>
    </MuiThemeProvider>
  ));

storiesOf('NavBar', module)
  .addDecorator(withTests('NavBar'))
  .addWithJSX('MORE', () => (
    <MuiThemeProvider theme={Theme}>
      <NavBar color="secondary">
        <NavBarItem label="Item 4" to="#Item4" />
        <NavBarItem label="Item 5" to="#Item5" />
        <NavBarItem label="Item 6" to="#Item6" />
        <NavBarItem label="Item 7" to="#Item7" />
        <NavBarItem label="Item 8" to="#Item8" />
      </NavBar>
    </MuiThemeProvider>
  ));
