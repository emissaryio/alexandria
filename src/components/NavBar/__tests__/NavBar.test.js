import React from 'react';
import { shallow } from 'enzyme';

import NavBar from '../NavBar';
import NavBarItem from '../NavBarItem';

describe('<NavBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <NavBar
        logo={{ src: 'https://placehold.it/259x28', alt: 'Logo' }}
      >
        <NavBarItem label="Item 1" to="#Item1" />
        <NavBarItem label="Item 2" to="#Item2" />
        <NavBarItem label="Item 3" to="#Item3" />
      </NavBar>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with className', () => {
    const wrapper = shallow(
      <NavBar
        logo={{ src: 'https://placehold.it/259x28', alt: 'Logo' }}
        className="custom-nav-bar"
      >
        <NavBarItem label="Item 1" to="#Item1" />
        <NavBarItem label="Item 2" to="#Item2" />
        <NavBarItem label="Item 3" to="#Item3" />
      </NavBar>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
