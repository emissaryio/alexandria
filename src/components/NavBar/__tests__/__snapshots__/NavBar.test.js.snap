// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders correctly 1`] = `
<div
  className=""
>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        href="/"
      >
        <pure(HomeIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Tabs)
        indicatorColor="primary"
        onChange={[Function]}
        value={0}
        variant="fullWidth"
      >
        <WithStyles(Tab)
          key=".0"
          label="Item 1"
        >
          <NavBarItem
            label="Item 1"
            to="#Item1"
          />
        </WithStyles(Tab)>
        <WithStyles(Tab)
          key=".1"
          label="Item 2"
        >
          <NavBarItem
            label="Item 2"
            to="#Item2"
          />
        </WithStyles(Tab)>
        <WithStyles(Tab)
          key=".2"
          label="Item 3"
        >
          <NavBarItem
            label="Item 3"
            to="#Item3"
          />
        </WithStyles(Tab)>
      </WithStyles(Tabs)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`<NavBar /> renders correctly with className 1`] = `
<div
  className="custom-nav-bar"
>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        href="/"
      >
        <pure(HomeIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Tabs)
        indicatorColor="primary"
        onChange={[Function]}
        value={0}
        variant="fullWidth"
      >
        <WithStyles(Tab)
          key=".0"
          label="Item 1"
        >
          <NavBarItem
            label="Item 1"
            to="#Item1"
          />
        </WithStyles(Tab)>
        <WithStyles(Tab)
          key=".1"
          label="Item 2"
        >
          <NavBarItem
            label="Item 2"
            to="#Item2"
          />
        </WithStyles(Tab)>
        <WithStyles(Tab)
          key=".2"
          label="Item 3"
        >
          <NavBarItem
            label="Item 3"
            to="#Item3"
          />
        </WithStyles(Tab)>
      </WithStyles(Tabs)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;
